src/ProjectTask.php

```diff
@@ -555,6 +561,52 @@ public function post_clone($source, $history)
         }
     }
 
+    /**
+     * Set automatically the real start and end dates if not set
+     * @param array $input // the input of the form
+     *
+     * @return array // the input with the real start and end dates set if needed
+     */
+    public function autoSetDate(array $input): array
+    {
+        if (!empty($input['real_start_date']) && !empty($input['real_end_date']) || empty($input['percent_done'])) {
+            return $input;
+        }
+        $real_start_date =  isset($input['real_start_date']) ?: isset($this->fields['real_start_date']) ?: null;
+        $real_end_date =  isset($input['real_end_date']) ?: isset($this->fields['real_end_date']) ?: null;
```

Comment:
`isset()` returns a boolean, so `$real_start_date` would contains a boolean or a null value.

```suggestion
        $real_start_date = ($input['real_start_date'] ?? null) ?: ($this->fields['real_start_date'] ?? null) ?: null;
        $real_end_date   = ($input['real_end_date'] ?? null) ?: ($this->fields['real_end_date'] ?? null) ?: null;
```